src/frequenz/sdk/microgrid/component_data.py

```diff
@@ -17,229 +19,218 @@
 from .component_states import EVChargerCableState
 
 
-class _BaseComponentData(ABC):
+@dataclass(frozen=True)
+class ComponentData(ABC):
```

Comment:
We don't need now the `ABC` here, right?

Reply:
I think we do, because it shouldn't be possible to create instances of `ComponentData`.

Reply:
BTW, if we really want to avoid people instantiating `ComponentData` it might be a good idea to add a test for it, so it doesn't accidentally break :)

Reply:
Good idea, I am adding a test. The good thing is that `mypy` will catch all instantiations of abstract classes:
```
************* Module tests.test_microgrid.test_component_data
tests/test_microgrid/test_component_data.py:20:8: E0110: Abstract class 'ComponentData' with abstract methods instantiated (abstract-class-instantiated)
```